lib/feature/main/main_screen.dart

```diff
@@ -1,22 +1,88 @@
 import 'package:netguru_flutter_template/app/s.dart';
+import 'package:netguru_flutter_template/feature/main/main_cubit.dart';
+import 'package:netguru_flutter_template/injection/injection.dart';
+import 'package:netguru_flutter_template/widget/custom_loading/custom_loading.dart';
+import 'package:netguru_flutter_template/values/app_theme.dart';
 
-class MainScreen extends StatelessWidget {
+class MainScreenRoute extends MaterialPageRoute<bool> {
+  MainScreenRoute() : super(builder: (_) => MainScreen());
+}
+
+class MainScreen extends StatefulWidget {
   @override
-  Widget build(BuildContext context) => Scaffold(
-        appBar: AppBar(title: Text(S.of(context).appName)),
-        body: Center(
-          child: Column(
-            mainAxisSize: MainAxisSize.min,
-            children: [
-              Text('Hello World!'),
-            ],
-          ),
-        ),
-      );
+  _MainScreenState createState() => _MainScreenState();
 }
 
-class MainScreenRoute extends MaterialPageRoute<bool> {
-  MainScreenRoute() : super(builder: (_) => MainScreen());
+class _MainScreenState extends State<MainScreen> {
+  MainCubit? mainCubit;
+
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    mainCubit = getIt.get<MainCubit>();
+    mainCubit?.fetchListItems();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<MainCubit>(
+      create: (context) => mainCubit!,
+      child: BlocConsumer<MainCubit, MainState>(
+        listener: (BuildContext context, state) {
+          setState(() {
+            isLoading = state is Loading;
+          });
+        },
+        builder: (BuildContext context, state) => Scaffold(
+          appBar: AppBar(
+            title: Text(S.of(context).appName),
+            backgroundColor: context.primaryColor(),
+          ),
+          body: Center(
+            child: Stack(
+              children: [
+                state is Loading ? CustomLoading() : SizedBox(),
+                state is Error ? Text(state.error.toString()) : SizedBox(),
+                if (state is Fetched)
+                  ListView(
+                      children: state.list.map(
+                    (s) {
+                      if (state.list.indexOf(s) % 2 == 0)
+                        s = 'I dont like even numbers';
+                      return Card(
+                        shape: RoundedRectangleBorder(
+                            borderRadius:
+                                BorderRadius.all(Radius.circular(20))),
+                        child: Container(
+                          color: Colors.green,
+                          child: Center(
+                            child: Padding(
+                              padding: EdgeInsets.all(8.0),
+                              child: Row(
+                                children: [
+                                  Image.asset(
+                                    'assets/images/splash_image.png',
```

Comment:
Also, consider creating an asset declaration file, you might need this image on another page and it would be extra work rewriting it each time, plus if the image in question changes you are forced to either retain the name by overwriting the previous image file or going to each file to adjust the name, However with the asset declaration file you can do it all from one spot and have it take change automatically in all the reference location